.github/workflows/ci.yml

```diff
@@ -231,7 +238,21 @@ jobs:
             echo "CARGO_INCREMENTAL=${{ env.CARGO_INCREMENTAL }}"
             echo "RUST_BACKTRACE=${{ env.RUST_BACKTRACE }}"
 
-      - uses: actions-rs/cargo@v1.0.3
+      - name: Check rustfmt
+        uses: actions-rs/cargo@v1.0.3
         with:
           command: fmt
           args: --all -- --check
+
+      # Edit zebra/deny.toml to allow duplicates
+      - name: Check for dependent crates with different versions
+        uses: EmbarkStudios/cargo-deny-action@v1.2.6
+        with:
+          command: check bans
+          args: --all-features --workspace
+
+      - name: Check crate sources
+        uses: EmbarkStudios/cargo-deny-action@v1.2.6
+        with:
+          command: check sources
+          args: --all-features --workspace
```

Comment:
Could this go in its own job in the CI workflow? It's not exactly of a piece of `rustfmt`

Reply:
We're having trouble with long CI run times. This is the most efficient place to put the duplicate crates and crate sources checks, because they just read `Cargo.lock`. (Like `rustfmt` reads Rust source files.) So these checks complete very quickly.

I tried renaming the job to "Rustfmt & Dependencies", but that broke GitHub's required checks. We can do the rename and change the GitHub config, but it would be easier when we have fewer PRs open.

I opened #2995 to review jobs and steps once CI is faster.